src/qcodes/parameters/multi_channel_instrument_parameter.py

```diff
@@ -45,16 +47,31 @@ def get_raw(self) -> tuple[ParamRawDataType, ...]:
         """
         return tuple(chan.parameters[self._param_name].get() for chan in self._channels)
 
-    def set_raw(self, value: ParamRawDataType) -> None:
+    def set_raw(self, value: ParamRawDataType | Sequence[ParamRawDataType]) -> None:
         """
-        Set all parameters to this value.
+        Set all parameters to this/these value(s).
 
         Args:
-            value: The value to set to. The type is given by the
+            value: The value(s) to set to. The type is given by the
                 underlying parameter.
         """
-        for chan in self._channels:
-            getattr(chan, self._param_name).set(value)
+        try:
+            for chan in self._channels:
+                getattr(chan, self._param_name).set(value)
+        except Exception as err:
+            try:
+                for chan, val in zip(self._channels, value, strict=True):
+                    getattr(chan, self._param_name).set(val)
```

Comment:
Might actually make more sense to check if value complies with zip(..., strict=True) before setting the parameters.

Reply:
I just realized that the strict kwarg has only been added in 3.10 (which gives rise to that weirdly misleading pyright error [here](https://github.com/microsoft/Qcodes/actions/runs/9147210941/job/25193293073#step:10:23). On the other hand, 3.9 support is due to be dropped in #5620.